test/extended/oauth/htpasswd.go

```diff
@@ -0,0 +1,87 @@
+package oauth
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Serial][Suite:openshift/oauth/serial] Htpasswd identity provider", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc                    = exutil.NewCLI("htpasswd-configure", exutil.KubeConfigPath())
+		testDataBaseDir       = exutil.FixturePath("testdata", "oauth_idp")
+		htpasswdFixture       = filepath.Join(testDataBaseDir, "oauth-htpasswd.yaml")
+		htpasswdSecretFixture = filepath.Join(testDataBaseDir, "htpasswd-secret.yaml")
+		cleanup               = func(ns string) {
+			err := oc.AsAdmin().KubeClient().CoreV1().Secrets("openshift-config").Delete("htpasswd", &metav1.DeleteOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			e2e.Logf("now resetting oauth cluster")
+			err = oc.AsAdmin().Run("delete").Args("crd", "oauths.config.openshift.io").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = wait.Poll(10*time.Second, 5*time.Minute, func() (done bool, err error) {
+				out, err := oc.AsAdmin().Run("get").Args("oauth", "cluster").Output()
+				if err != nil {
+					e2e.Logf("oauth crd does not exist")
+					return false, nil
+				}
+				if !strings.Contains(out, "cluster") {
+					return false, nil
+				}
+				return true, nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			// cleanup created users/identity
+			testuser := fmt.Sprintf("%s-user", ns)
```

Comment:
where does this user come from? as far as I could tell from the htpasswd secret that's provided in this patch; the user shouldn't have the namespace/project as a prefix.

Reply:
:woman_shrugging: this user is created as part of the test framework (I'll have to dig to explain any further), I only know this from testing experience, ie after running the test `oc get users` shows that.

Reply:
asked around, i think i'll only clean up the user/identity this test creates, not the users created via framework (the namespace-user).